.pre-commit-config.yaml

```diff
@@ -0,0 +1,15 @@
+repos:
+  - repo: local
+    hooks:
+    - id: codechecks
+      name: codechecks
+      stages: ["commit"]
+      entry: "tox -e codechecks"
+      language: system
+      pass_filenames: false
+    - id: test
+      name: test
+      stages: ["push"]
+      entry: "make test"
```

Comment:
Does defining this in terms of a "make" invocation generally work on MacOS and Windows? (That is, do those platforms typically have "make" installed?)

Reply:
For use as a hook we want it to [fail early](https://github.com/tahoe-lafs/tahoe-lafs/pull/830/commits/2ef422e064c430da43919342bf363959bef8e67b#diff-b67911656ef5d18c4ae36cb6741b7965R41) and [run as fast as possible](https://github.com/tahoe-lafs/tahoe-lafs/pull/830/commits/2ef422e064c430da43919342bf363959bef8e67b#diff-b67911656ef5d18c4ae36cb6741b7965R43).  That requires 2 tox commands and is also useful outside of the VCS hooks, hence placing it in the `./Makefile` as a readable way to do that.

In terms of what's typical, I've seen make used in `./.pre-commit-config.yaml` on several other projects and many of the OS X developers I know have GNU make installed and understand how and when to use it.

I haven't gotten the impression while getting up to speed on the state of things in the code base that there is a "blessed" development environment at all, much less that the cross-platform compatibility of any such set up is maintained.  Am I missing that somewhere?

Reply:
I don't know that's written down anywhere, but we do support Windows and Mac as targets. I don't know that we *have* any windows-based developers right now, but it would be nice :)

Reply:
Ok, nice sure, but couldn't (maybe shouldn't) "improve windows development" story be a separate PR and not necessarily hold up this one?